voxblox/src/core/esdf_map.cc

```diff
@@ -50,4 +50,123 @@ bool EsdfMap::isObserved(const Eigen::Vector3d& position) const {
+    gradients.col(i) = gradient;
+  }
+}
+
+void EsdfMap::batchIsObserved(
+          EigenDRef<const Eigen::Matrix<double, 3, Eigen::Dynamic>>& positions,
+          Eigen::Ref<Eigen::VectorXi> observed) const {
+  /* TODO(mereweth@jpl.nasa.gov) - this looks like it gets truncated on return
+   * to Python anyway. Throw std::runtime_error here if too small?
+   */
+  observed.resize(positions.cols());
+  for (int i = 0; i < positions.cols(); i++) {
+    observed[i] = isObserved(positions.col(i));
+  }
+}
+
+unsigned int EsdfMap::coordPlaneSliceGetDistance(
+                unsigned int free_plane_index,
+                double free_plane_val,
+                EigenDRef<Eigen::Matrix<double, 3, Eigen::Dynamic>>& positions,
+                Eigen::Ref<Eigen::VectorXd> distances) const {
+  BlockIndexList blocks;
+  esdf_layer_->getAllAllocatedBlocks(&blocks);
+
+  // Cache layer settings.
+  size_t vps = esdf_layer_->voxels_per_side();
+  size_t num_voxels_per_block = vps * vps * vps;
+
+  // Temp variables.
+  bool did_all_fit = true;
+  unsigned int count = 0;
+
+  // Iterate over all blocks.
+  for (const BlockIndex& index : blocks) {
+    // Iterate over all voxels in said blocks.
+    const Block<EsdfVoxel>& block = esdf_layer_->getBlockByIndex(index);
+
+    Point origin = block.origin();
+    if (std::abs(origin(free_plane_index) - free_plane_val)
+                                        > block.block_size()) {
+                                        //> block.block_size() / 2.0) {
+                                        //> esdf_layer_->block_size() / 2.0) {
+      continue;
+    }
+
+    for (size_t linear_index = 0; linear_index < num_voxels_per_block;
+         ++linear_index) {
+      Point coord = block.computeCoordinatesFromLinearIndex(linear_index);
+      const EsdfVoxel& voxel = block.getVoxelByLinearIndex(linear_index);
+      if (std::abs(coord(free_plane_index) - free_plane_val)
+                                          <= block.voxel_size()) {
+                                          //<= block.voxel_size() / 2.0) {
+                                          //<= esdf_layer_->voxel_size() / 2.0) {
+        double distance;
+        if (voxel.observed) {
+          distance = voxel.distance;
+        }
+        else {
+          continue;
+          // TODO(mereweth@jpl.nasa.gov) - how to indicate unobserved?
+          //distance = 0;
+        }
+
+        // TODO(mereweth@jpl.nasa.gov) - implement max points to return
+        if (count < positions.cols()) {
+          positions.col(count) = Eigen::Vector3d(coord.x(), coord.y(), coord.z());
+        }
+        else {
+          did_all_fit = false;
+        }
+        if (count < distances.size()) {
+          distances(count) = distance;
+        }
+        else {
+          did_all_fit = false;
+        }
+        count++;
+      }
+    }
+  }
+
+  if (!did_all_fit) {
+    throw std::runtime_error(std::string("Unable to store ") + std::to_string(count) + " values.");
```

Comment:
Pybind11 converts std::exceptions into equivalent Python exceptions, which can be caught in Python. This seems pretty Pythonic, and I tried to only add exceptions where I was creating a new function specifically to bind in Python. Would that be OK to add exceptions in that specific case?

Does this function seem like something you'd like to use (aside from generating Python bindings?)